src/Glpi/System/Log/LogParser.php

```diff
@@ -250,12 +250,11 @@ public function getFullPath(string $filepath): ?string
     {
         $filepath = str_replace('\\', '/', $filepath);
 
-        if (preg_match('/\/..\//', $filepath) === 1) {
+        $fullpath = realpath($this->directory . '/' . $filepath); //@phpstan-ignore theCodingMachineSafe.function (false is explicitly tested)
+        if ($fullpath === false || !str_starts_with($fullpath, $this->directory)) {
```

Comment:
In a few cases, `GLPI_LOGS_DIR` realpath can be different than the constant value. To prevvent issues, it is preferable to apply the `realpath` transformation to `$this->directory` before checking path safeness with `str_starts_with()`.

```suggestion
        $logs_dir_path = realpath($this->directory); //@phpstan-ignore theCodingMachineSafe.function (false is explicitly tested)
        if ($logs_dir_path === false) {
            return null;
        }
        
        $fullpath = realpath($logs_dir_path . '/' . $filepath); //@phpstan-ignore theCodingMachineSafe.function (false is explicitly tested)
        if ($fullpath === false || !str_starts_with($fullpath, $logs_dir_path)) {
```